addons/website_sale_loyalty/controllers/main.py

```diff
@@ -23,8 +23,10 @@ def pricelist(self, promo, **post):
             reward_successfully_applied = True
             if len(coupon_status) == 1:
                 coupon, rewards = next(iter(coupon_status.items()))
-                if len(rewards) == 1 and not rewards.multi_product:
+                if len(rewards) == 1 and rewards.reward_type != 'product':
                     reward_successfully_applied = self._apply_reward(order, rewards, coupon)
+                elif product:
+                    reward_successfully_applied = self._apply_reward(order, rewards, coupon, product=product)
```

Comment:
```suggestion
                if len(rewards) == 1 and (product or not rewards.multi_product):
                    reward_successfully_applied = self._apply_reward(
                        order, rewards, coupon, product=product,
                    )
```
Should work :thinking: